Application/Features/Queries/ShowAvailableReservations/ShowAvailableReservationsQueryHandler.cs

```diff
@@ -36,15 +36,18 @@ public class ShowAvailableReservationsQueryHandler : IRequestHandler<ShowAvailab
 
         var availableReservations = new List<AvailableReservationDto>();
 
-        foreach (var reservation in reservations)
+        foreach (var movie in movies)
         {
-            var movie = movies.FirstOrDefault(x => x.Id == reservation.MovieId);
-            if (movie is not null)
+            var reservationList = reservations.FindAll(x => x.MovieId == movie.Id);
+            if (reservationList.Count > 0)
             {
-                availableReservations.Add(AvailableReservationDto.FromMovieReservation(movie, reservation));
+                availableReservations.AddRange(reservationList.Select(reservation => AvailableReservationDto.FromMovieReservation(movie, reservation)));
             }
         }
 
-        return availableReservations;
+        return availableReservations
```

Comment:
Ovde imamo paging implementiran na nivou Availability servisa. Problem je što on i dalje dovlači sve filmove i sve rezervacije, tako da taj deo može da se optimizuje. Rezervacije svakako nećemo više da dovlačimo sve, već samo po filmu, a za filmove treba da propagiramo paging